### PR TITLE
[Trivial] Some code style changes

### PR DESCRIPTION
**Pull request by aparatext**
This PR mostly just changes some trivial code formatting. I hope it would be useful. A lot of it is subjective, but at times I just felt it looked better. I'm not trying to impose my own preference on you, so you are free to make any necessary changes.

It also does the following:
 - Adds `clippy` and `rustfmt` to the pipeline. I'm not sure if the pipeline has the a write permission or if I've done it correctly, so it might need to be replaced with the proper action.
 - Formats the code with `rustfmt` and applies some `clippy` suggestions. I've also added some line breaks to make it look less dense. I hope they make sense and separate different things correctly.
 - Fixes some Markdown formatting.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,69 +1,88 @@
...
+## Legacy queries
+ - `get_video`: Fetches all information about the video except captions and storyboards
+ - `get_channel_info`: Fetches all channel information and about tab
+ - `get_channel_tab`: Fetches a specific tab like videos to get channel videos
+ - `search`: Search YouTube
+ - `load_search`: Continue search with ctoken
+ - `get_comments`: Loads initial comments or more comments for video
+ - `load_related_videos`: Loads the related videos
+ - `get_playlist`: Loads a playlist
+
```
Comment (AndromedaCLI) [inline README.md]: Could you while you're at it remove the legacy queries part since nobody should be using those, would be nice

**Comment by AndromedaCLI** [comment]
All good I wanted to do that pipeline anyways so I appreciate it :)

File: README.md (hunk 1 above)
Comment (aparatext) [reply README.md]: Should I remove all legacy queries? Are they going to be removed from the codebase?

Comment (AndromedaCLI) [reply README.md]: Yeah all, since they are already removed in dev but not main(released)

**Comment by aparatext** [comment]
> All good I wanted to do that pipeline anyways so I appreciate it :)

I think the formatting should be migrated to a separate into GitHub Action to work properly.

**Comment by AndromedaCLI** [comment]
Well should be fine for now

**Comment by aparatext** [comment]
This may not be a place to discuss this, but should I make a second PR to replace some unwraps with error propagation?

**Comment by AndromedaCLI** [comment]
Sorry for the late answer didn't saw it here but yeah sure, I had an idea on how to improve the current api by alot tho so it would mean we don't need to write a new parser :)

**Pull request merged**